### PR TITLE
build(deps): reduce CVEs in dependencies and runtime environment (PROJQUAY-4777)

### PR DESCRIPTION
**Pull request by dmesser**
- switch to ubi-minimal to get rid of subscription-manager and associated old urllib3 versions
- remove older pip version part of python39 appstream
- bump python dependencies

Signed-off-by: dmesser <dmesser@redhat.com>

### CODE REVIEW

**Review by flavianmissi — APPROVED** [review]
LGTM.

we need a plan to propagate these changes downstream - if I remember correcty that doesn't happen automatically. cc @HammerMeetNail

**Comment by HammerMeetNail** [comment]
Yeah, the `requirements.txt` might be good for downstream, but the `Dockerfile` changes need to be pulled in. We can take a look at this Monday.

**Comment by dmage** [comment]
Why can't we use pip from python39 appstream?

**Comment by dmesser** [comment]
Causes false positives with Clair scanning and is generally older than upstream.

**Comment by flavianmissi** [comment]
@dmesser looks like this needs a rebase

File: Dockerfile [hunk 1]
```diff
@@ -45,21 +48,22 @@ RUN set -ex\
...
         file \
         make \
         libjpeg-turbo \
         libjpeg-turbo-devel \
-		wget\
-	; dnf -y -q clean all
+		wget \
+	; microdnf -y clean all
 WORKDIR /build
+RUN python3 -m ensurepip --upgrade
```
Comment (dmage) [inline Dockerfile:66]: Downstream build system doesn't allow us to download unpinned versions. Most likely it'll be removed when we reconcile upstream and downstream Dockerfile (ideally we should be able to generate downstream Dockerfile with `sed`)

Comment (dmesser) [reply Dockerfile:66]: Isn't ensurepip part of python and hence the pip version is aligned to the python version?

Comment (dmage) [reply Dockerfile:66]: TIL (from the front documentation page):

> This module (ensurepip) does not access the internet. All of the components needed to bootstrap pip are included as internal parts of the package.

**Review by dmage — APPROVED** [review]

**Pull request merged**